Sources/MeiliSearch/Async/Client+async.swift

```diff
@@ -0,0 +1,213 @@
+      self.waitForTask(task: task, options: options) { result in
+        continuation.resume(with: result)
+      }
+    }
+  }
+
+  /**
+   See `getTask(taskUid:_:)`
+   */
+  public func getTask(taskUid: Int) async throws -> Task {
+    try await withCheckedThrowingContinuation { continuation in
+      self.getTask(taskUid: taskUid) { result in
+        continuation.resume(with: result)
+      }
+    }
+  }
+
+  /**
+   See `getTasks(params:_:)`
+   */
+  public func getTasks(params: TasksQuery? = nil) async throws -> TasksResults {
+    try await withCheckedThrowingContinuation { continuation in
+      self.getTasks(params: params) { result in
+        continuation.resume(with: result)
+      }
+    }
+  }
+
+  /**
+   See `getKeys(params:_:)`
+   */
+  public func getKeys(params: KeysQuery? = nil) async throws -> KeysResults {
+    try await withCheckedThrowingContinuation { continuation in
+      self.getKeys(params: params) { result in
+        continuation.resume(with: result)
+      }
+    }
+  }
+
+  /**
+   See `getKey(keyOrUid:_:)`
+   */
+  public func getKey(keyOrUid: String) async throws -> Key {
+    try await withCheckedThrowingContinuation { continuation in
+      self.getKey(keyOrUid: keyOrUid) { result in
+        continuation.resume(with: result)
+      }
+    }
+  }
+
+  /**
+   See `createKey(_:_:)`
+   */
+  public func createKey(_ keyParams: KeyParams) async throws -> Key {
+    try await withCheckedThrowingContinuation { continuation in
+      self.createKey(keyParams) { result in
+        continuation.resume(with: result)
+      }
+    }
+  }
+
+  /**
+   See `updateKey(keyOrUid:keyParams:_:)`
+   */
+  public func updateKey(keyOrUid: String, keyParams: KeyUpdateParams) async throws -> Key {
+    try await withCheckedThrowingContinuation { continuation in
+      self.updateKey(keyOrUid: keyOrUid, keyParams: keyParams) { result in
+        continuation.resume(with: result)
+      }
+    }
+  }
+
+  /**
+   See `deleteKey(keyOrUid:_:)`
+   */
+  public func deleteKey(keyOrUid: String) async throws {
+    try await withCheckedThrowingContinuation { continuation in
+      self.deleteKey(keyOrUid: keyOrUid) { result in
+        continuation.resume(with: result)
+      }
+    }
+  }
+
+  /**
+   See `allStats(_:)`
+   */
+  public func allStats() async throws -> AllStats {
+    try await withCheckedThrowingContinuation { continuation in
+      self.allStats { result in
+        continuation.resume(with: result)
+      }
+    }
+  }
+
+  /**
+   See `health(_:)`
+   */
+  public func health() async throws -> Health {
+    try await withCheckedThrowingContinuation { continuation in
+      self.health { result in
+        continuation.resume(with: result)
+      }
+    }
+  }
+
+  /**
+   See `isHealthy(_:)`
+   */
+  public func isHealthy() async -> Bool {
+    await withCheckedContinuation { continuation in
+      self.isHealthy { result in
+        continuation.resume(returning: result)
+      }
+    }
+  }
```

Comment:
This implementation is very slightly different to all others - this is because the underlying implementation does not return a result, but rather returns `false` if an error is identified.

As such we use a checked continuation (no throwing), and forward the bool response verbatim.

Works the same, just no throwing.

Reply:
Nice.

However this does point to an important omission that I would say blocks this PR currently - the lack of tests.

I think one way to test things might be to check that the async results are equivalent to the non-async ones - this would avoid duplicating tests but still ensure our assumptions about success, failure, and the compatibility with the old-style api.

(If you have time cool, if not I can probs take a look at it as well in the next few days.)